python/pyspark/sql/tests.py

```diff
@@ -1178,6 +1183,74 @@ def test_functions_broadcast(self):
         # planner should not crash without a join
         broadcast(df1)._jdf.queryExecution().executedPlan()
 
+    def test_basic_typed_operations(self):
```

Comment:
Maybe it will be easier to reason about test failures if we break it to multiple `def`s?